ouroboros-network/src/Ouroboros/Network/Diffusion/P2P.hs

```diff
@@ -868,328 +862,324 @@ runM Interfaces
-                  -- peer state actions
-                  --
-                  -- Peer state actions run a job pool in the background which
-                  -- tracks threads forked by 'PeerStateActions'
+                  -- Run peer selection (p2p governor)
                   --
 
-                  withPeerStateActions
-                    PeerStateActionsArguments {
-                        spsTracer = dtPeerSelectionActionsTracer,
-                        spsDeactivateTimeout = Diffusion.Policies.deactivateTimeout,
-                        spsCloseConnectionTimeout =
-                          Diffusion.Policies.closeConnectionTimeout,
-                        spsConnectionManager = connectionManager,
-                        spsExitPolicy = stdExitPolicy daReturnPolicy
-                      }
-                    $ \(peerStateActions
-                          :: NodeToNodePeerStateActions
-                               InitiatorResponderMode ntnAddr versionData m a ()) ->
-
-                    --
-                    -- Run peer selection (p2p governor)
-                    --
-
-                    withPeerSelectionActions
-                      dtTraceLocalRootPeersTracer
-                      dtTracePublicRootPeersTracer
-                      diNtnToPeerAddr
-                      diLocalAndPublicRootDnsSemaphore
-                      (diDnsActions lookupReqs)
-                      (readTVar peerSelectionTargetsVar)
-                      daReadLocalRootPeers
-                      daReadPublicRootPeers
-                      daOwnPeerSharing
-                      (pchPeerSharing diNtnPeerSharing)
-                      (readTVar (getPeerSharingRegistry daPeerSharingRegistry))
-                      (readMessage outboundInfoChannel)
-                      peerStateActions
-                      requestLedgerPeers
-                      $ \localPeerRootProviderThread
-                        (peerSelectionActions
-                           :: NodeToNodePeerSelectionActions
-                                InitiatorResponderMode ntnAddr versionData m a ()) ->
+                  withPeerSelectionActions
+                    dtTraceLocalRootPeersTracer
+                    dtTracePublicRootPeersTracer
+                    dtTraceLedgerPeersTracer
+                    diNtnToPeerAddr
+                    (diDnsActions lookupReqs)
+                    (readTVar peerSelectionTargetsVar)
+                    daReadLocalRootPeers
+                    daReadPublicRootPeers
+                    daOwnPeerSharing
+                    (pchPeerSharing diNtnPeerSharing)
+                    (readTVar (getPeerSharingRegistry daPeerSharingRegistry))
+                    (readMessage outboundInfoChannel)
+                    peerStateActions
+                    ledgerPeersRng
+                    daLedgerPeersCtx
+                    daReadUseLedgerAfter
+                    $ \(ledgerPeersThread, localRootPeersProvider)
```

Comment:
```suggestion
                    $ \(ledgerPeersThread, localRootPeersProviderThread)
```
If we keep the old name it will be more consistent, and the diff will be cleaner.